File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    private List<int> _scoresList { get; }

    public HighScores(List<int> list) => _scoresList = list;

    public List<int> Scores() => _scoresList;

    public int Latest() => _scoresList.Last();

    public int PersonalBest() => _scoresList.Max();

    public List<int> PersonalTopThree()
        => _scoresList.OrderDescending().Take(3).ToList();
}

File: fsharp/high-scores/HighScores.fs
module HighScores

let scores (values: int list) : int list = values

let latest (values: int list) : int = List.last values

let personalBest (values: int list) : int = List.head (List.sortDescending values)

let personalTopThree (values: int list) : int list =
    List.truncate 3 (List.sortDescending values)
